region/brinstar/blue/Morph Ball Room.json

```
@@ -241,6 +241,45 @@
         "Using the full runway, this requires a last-frame jump, followed by a 2-frame window for the morph."
       ]
     },
+    {
+      "link": [1, 1],
+      "name": "R-Mode Spark Interrupt (Gain Blue Suit)",
+      "entranceCondition": {
+        "comeInWithRMode": {}
+      },
+      "requires": [
+        "h_ZebesIsAwake",
+        {"or": [
+          {"enemyDamage": {"enemy": "Sidehopper", "type": "contact", "hits": 1}},
+          "ScrewAttack"
+        ]},
+        {"or": [
+          "h_CrystalFlash",
+          {"and":[
+            {"disableEquipment": "ETank"},
+            {"resourceAvailable": [{"type": "Energy", "count": 95}]},
+            {"resourceMissingAtMost": [{"type": "Missile", "count": 0}]},
+            {"resourceMissingAtMost": [{"type": "Super", "count": 0}]},
+            {"partialRefill": {"type": "ReserveEnergy", "limit": 1}},
+            "canBePatient"
+          ]}
+        ]},
+        "canTrickyDodgeEnemies",
+        "canShinechargeMovementComplex",
+        {"canShineCharge": { "usedTiles": 25, "openEnd": 0 } },
+        {"autoReserveTrigger": { "maxReserveEnergy": 95 }},
+        "canRModeSparkInterrupt"
+      ],
+      "clearsObstacles": ["C"],
+      "flashSuitChecked": true,
+      "blueSuitChecked": true,
+      "devNote": [
+        "Due to R-Mode Entry, getting hit by the Sidehopper can only be avoided by a quick Screw Attack after releasing the X-Ray Scope.",
+        "With Screw Attack you can attempt to farm up to two Sidehoppers for energy. Otherwise, you must use a Crystal Flash.",
```

```suggestion
        "You can attempt to farm up to two Sidehoppers for Reserve Energy. Otherwise, you must use a Crystal Flash.",
```
Farming can also be applicable to non-Screw options, where you take damage.